Calculator/Calculator/Model/Formula.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  Formula.swift
+//  Calculator
+//
+//  Created by leewonseok on 2022/09/20.
+//
+
+import Foundation
+
+struct Formula {
+    var operands = CalculatorItemQueue()
+    var operators = CalculatorItemQueue()
+    
+    public mutating func result() throws -> Double {
+        guard var totalNumber = operands.dequeue() as? Double else { throw CalculatorError.emptyError }
+
+        for _ in 1...operands.count {
+            guard let number = self.operands.dequeue() as? Double else { throw CalculatorError.emptyError }
+            guard let operatorSymbol = self.operators.dequeue() as? Operator else { throw CalculatorError.emptyError }
+            
+            if operatorSymbol == Operator.divide && number == 0.0 { throw CalculatorError.divideError }
+            
+            totalNumber = operatorSymbol.calculate(lhs: totalNumber, rhs: number)
+        }
+        return totalNumber
+    }
```

Comment:
```suggestion
    public mutating func result() throws -> Double {
        guard var result = operands.dequeue() as? Double else {
            throw CalculatorError.emptyError
        }

        while operands.isEmpty == false {
            guard let rhs = self.operands.dequeue() as? Double,
                  let `operator` = self.operators.dequeue() as? Operator
            else {
                throw CalculatorError.emptyError
            }

            if `operator` == .divide && rhs == .zero {
                throw CalculatorError.divideError
            }
            
            result = `operator`.calculate(lhs: result, rhs: rhs)
        }
        return result
    }
```

이런식으로 하는건 어떨까요?
줄바꿈이나 네이밍에 개인적인 취향차이가 있을 수 있기 때문에 반드시 수정할 필요는 없습니다!
다만, 저는 개인적으로 봤을때 한눈에 코드가 파악되지 않더라고요🫠

Reply:
추가로 public으로 설정한 이유가 있나요??

Reply:
![](https://i.imgur.com/ZmLOrpr.png)
 클래스 다이어그램에서 '+' 기호는 public이라고 명시되어있어서 UML상의 기호를 따라 작성했습니다!

Reply:
오...이번에도 네이밍에 대해서 배워갑니다..!!
`operator`같이 사용 하는군요..!!
number == 0.0보다 .zero가 더 가독성도 좋아보이고 눈에 확 들어오네요..!!
리뷰 반영하였습니다!! 감사합니다

Reply:
저희가 package나 protected같은 접근제어자가 있나요? 🫣